proposals/20230530-driver-kernel-testing-framework.md

```diff
@@ -0,0 +1,454 @@
+# Driver Kernel Testing Framework
+
+## Objective
+
+The objective is to establish a comprehensive and structured testing process for Falco's kernel drivers. 
+
+This framework ensures compatibility, stability, and optimal performance across different kernel versions, distributions, and architectures. The intended outcome is a robust testing infrastructure, enabling developers and adopters to have confidence in the reliability and functionality of the kernel drivers.
+
+Additionally, it aims to set clear expectations and increase awareness among adopters by consistently communicating the complexities involved in kernel testing.
+
+
+## Key Success Indicators
+
+Acceptable confidence in the kernel drivers can be evaluated based on the following factors:
+
+- Functionality: The drivers should run and work as intended, effectively capturing events within the kernel environment for the purpose of threat detection.
+
+- Regression testing: Core metrics, which will be determined, should be monitored to ensure that there are no significant regressions or performance issues introduced with each iteration or update of the kernel drivers.
+
+- Cost budgeting: The cost associated with kernel monitoring for threat detection should be appropriately budgeted, taking into consideration factors such as resource utilization, scalability, and the overall value provided by the monitoring solution.
+
+By assessing these indicators, we can gauge the overall confidence, success, and performance of the kernel drivers within [The Falco Project](https://falco.org/).
+
+The initial proposal primarily emphasizes the "Functionality" aspect, while the other key indicators will be addressed in future iterations.
+
+## Key Terms
+
+First, let's clarify a few definitions and provide further context.
+
+- `kernel versions`: In the context of the testing framework, kernel versions refer to changes in the major and minor version of the kernel (e.g., 5.15 or 6.4). These version changes are specifically relevant for testing the Falco drivers, with a particular emphasis on testing with Long-Term Support (LTS) releases.
```

Comment:
i'd use `kernel releases` here; that's what we use elsewhere, and what test-infra dbg, driverkit and kernel-crawler use.